packages/compass-logging/src/index.ts

```diff
@@ -43,8 +45,30 @@ export function createLoggerAndTelemetry(component: string): {
   } as Writable;
   const writer = new MongoLogWriter('', null, target);
 
-  const track = (event: string, properties: Record<string, any> = {}): void => {
-    emit(ipc, 'compass:track', { event, properties });
+  const track = (...args: [string, TrackProps?]) => {
+    void Promise.resolve()
+      .then(() => trackAsync(...args))
+      .catch((error) => debug('track failed', error));
+  };
+
+  const trackAsync = async (
+    event: string,
+    properties: TrackProps = {}
+  ): Promise<void> => {
+    const isTrackingEnabled = (global as any)?.hadronApp?.isFeatureEnabled(
+      'trackUsageStatistics'
+    );
+    if (!isTrackingEnabled) {
+      return;
+    }
+    const data = {
+      event,
+      properties,
+    };
+    if (typeof properties === 'function') {
```

Comment:
just a super non important detail which here doesn't even matter for how simple this function is. I find out cleaner and safer to avoid mutating variables when possible:

``` js
const data = {
  event,
  properties: typeof properties === 'function' ?
    await properties() : properties
};

emit(ipc, 'compass:track', data);
```